pkg/sparrow/controller.go

```diff
@@ -0,0 +1,226 @@
+// sparrow
+// (C) 2024, Deutsche Telekom IT GmbH
+//
+// Deutsche Telekom IT GmbH and all other contributors /
+// copyright owners license this file to you under the Apache
+// License, Version 2.0 (the "License"); you may not use this
+// file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package sparrow
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/caas-team/sparrow/internal/logger"
+	"github.com/caas-team/sparrow/pkg/checks"
+	"github.com/caas-team/sparrow/pkg/checks/runtime"
+	"github.com/caas-team/sparrow/pkg/db"
+	"github.com/getkin/kin-openapi/openapi3"
+)
+
+// ChecksController is responsible for managing checks.
+type ChecksController struct {
+	db      db.DB
+	metrics Metrics
+	checks  runtime.Checks
+	cErr    chan error
+	done    chan struct{}
+}
+
+// NewChecksController creates a new ChecksController.
+func NewChecksController(dbase db.DB, metrics Metrics) *ChecksController {
+	return &ChecksController{
+		db:      dbase,
+		metrics: metrics,
+		checks:  runtime.Checks{},
+		cErr:    make(chan error, 1),
+		done:    make(chan struct{}, 1),
+	}
+}
+
+// ListenErrors listens for errors and unregisters failed checks.
+func (cc *ChecksController) ListenErrors(ctx context.Context) {
```

Comment:
HandleErrors feels like a more appropriate name. We're not listening to a socket or a connection of some kind. We just handling errors as they come in through the channel.